community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/LoadCsvPeriodicCommitObserver.scala

```diff
@@ -43,8 +45,8 @@ class LoadCsvPeriodicCommitObserver(batchRowCount: Long, resources: ExternalReso
   }
 
   private def onNext() {
-    updateCounter += 1
     updateCounter.resetIfPastLimit(batchRowCount)(commitAndRestartTx())
+    updateCounter += 1
```

Comment:
This change is what makes `PERIODIC COMMIT` commit _after_ the specified number of rows have been processed, rather than _just before_ the specified number of rows is fed into the pipeline.